lua/mason-registry/selene/init.lua

```diff
@@ -1,11 +1,34 @@
 local Pkg = require "mason-core.package"
-local cargo = require "mason-core.managers.cargo"
+local _ = require "mason-core.functional"
+local platform = require "mason-core.platform"
+local github = require "mason-core.managers.github"
+local std = require "mason-core.managers.std"
+
+local coalesce, when = _.coalesce, _.when
 
 return Pkg.new {
     name = "selene",
     desc = [[A blazing-fast modern Lua linter written in Rust]],
     homepage = "https://kampfkarren.github.io/selene/",
     languages = { Pkg.Lang.Lua },
     categories = { Pkg.Cat.Linter },
-    install = cargo.crate("selene", { bin = { "selene" } }),
+    ---@async
+    ---@param ctx InstallContext
+    install = function(ctx)
+        github
+            .unzip_release_file({
+                repo = "Kampfkarren/selene",
+                asset_file = function(release)
+                    local target = coalesce(
+                        when(platform.is.mac_x64, "selene-%s-macos.zip"),
```

Comment:
This runs fine on arm macs too!
```suggestion
                        when(platform.is.mac, "selene-%s-macos.zip"),
```